src/Microsoft.DotNet.VersionTools/Dependencies/ProjectJsonUpdater.cs

```diff
@@ -29,33 +29,20 @@ public IEnumerable<BuildInfo> Update(IEnumerable<BuildInfo> buildInfos)
         {
             foreach (string projectJsonFile in ProjectJsonPaths)
             {
-                JObject projectRoot;
+                IEnumerable<BuildInfo> buildInfosUsed = null;
+
                 try
                 {
-                    projectRoot = ReadProject(projectJsonFile);
+                    FileUtils.ReplaceFileContents(projectJsonFile, contents => ReplaceAllDependencyVersions(contents, projectJsonFile, buildInfos, out buildInfosUsed));
                 }
                 catch (Exception e)
                 {
-                    Trace.TraceWarning($"Non-fatal exception occurred reading '{projectJsonFile}'. Skipping file. Exception: {e}. ");
+                    Trace.TraceWarning($"Non-fatal exception occurred reading or writing '{projectJsonFile}'. Skipping file. Exception: {e}. ");
```

Comment:
I'm thinking about https://github.com/naamunds/buildtools/blob/85446ddcbb1f7d15ee4491b5fea3d5266b69aaee/src/Microsoft.DotNet.VersionTools/Dependencies/ProjectJsonUpdater.cs#L87-L118. I think it wouldn't be hard to craft some json that crashes that. (And I wouldn't classify it as reading or writing.)